cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -371,6 +412,30 @@ private void getAppDetails(HttpResponder responder, String namespace, String nam
     }
   }
 
+  // parse a filename into an Id.Artifact, assuming the filename is of format {name}-{version}.jar
```

Comment:
Will this be the only way to get the version of the jar? Will we also have a fall back option of trying to get it from the manifest?